apps/digital-inclusion/behaviours/save-data.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const mysql = require('mysql');
+const config = require('../../../config');
+
+const pool = mysql.createPool({
+  host: config.sql.host,
+  user: config.sql.user,
+  password: config.sql.password,
+  database: config.sql.db
+});
+
+module.exports = superclass => class extends superclass {
+  saveValues(req, res, next) {
+    const project = req.sessionModel.get('project');
+    const researchDate = req.sessionModel.get('research-date');
+    const category = req.sessionModel.get('category');
+    const accessNeeds = req.sessionModel.get('access-needs');
+
+    // build sql statement to deal with empty researchDate or/and accessNeeds,
+    // this will mean a null will go into the db
+    // bug: mysql needs insert data to be wrapped in quotations '' even though it is a string
+    let sql = `INSERT INTO research_data SET project='${project}', category='${category}'`;
```

Comment:
I would farm this work out into its own module. (Probably you would call that the 'model' or something.)

That way it will be easier to test as you have less of a dance to do with instantiating the ES6 class.

Reply:
Your 'model' module would probably offer CRUD (Create, Read, Update & Delete) functionality plus maybe 'Search'. (Though don't bother implementing anything you don't currently need.)

So the code here would just do something like:
```
const model = require('some/path/model/digital-inclusion');

module.exports = superclass => class extends superclass {
  saveValues(req, res, next) {
    model.create({
      project: req.values.project,
      category: req.values.category
    });
  }
};
```

Reply:
That's how I would do it anyway.

Now on to the actual SQL that you have here...

First of all your SQL syntax is wrong, take a look at https://dev.mysql.com/doc/refman/5.7/en/insert.html to see how to do an INSERT in MySQL.

Also the way that you are putting the form data into the SQL is _extremely_ dangerous. Read up on 'SQL injection' to see what I mean. (Give me a shout if it doesn't make sense.)

Reply:
I should add that modern libraries generally provide a way of safely doing this out of the box.